src/new-components/checkbox-field/checkbox-field.stories.mdx

```diff
@@ -54,20 +54,38 @@ export const Template = ({
     disabled,
     indeterminate,
     label,
+    icon,
     'aria-controls': ariaControls,
     'aria-describedby': ariaDescribedby,
     'aria-label': ariaLabel,
     'aria-labelledby': ariaLabelledby,
-}) => <CheckboxField disabled={disabled} indeterminate={indeterminate} label={label} />
+}) => <CheckboxField disabled={disabled} indeterminate={indeterminate} label={label} icon={icon} />
 
-### ReactNode as Label
+### CheckboxField with icon
+
+A checkbox field with an icon and a label.
+
+<Canvas>
+    <Story
+        parameters={{ docs: { source: { type: 'dynamic' } } }}
+        name="CheckboxField with icon"
+        args={{
+            label: 'Label with icon',
+            icon: <Icon />,
+        }}
+    >
+        {Template.bind({})}
+    </Story>
+</Canvas>
+
+### ReactNode as Label (deprecated)
 
 A checkbox field with an icon and a string for the built-in label element.
 
 <Canvas>
     <Story
         parameters={{ docs: { source: { type: 'dynamic' } } }}
-        name="ReactNode as Label"
+        name="ReactNode as Label (deprecated)"
```

Comment:
Err I'm now thinking we don't actually want to discourage passing a `ReactNode` into the label at all... What I meant here is that this is a deprecated way to pass an icon to the `CheckboxField` element, since now we have `icon`. I'll try to rephrase these stories a little bit. Please hold.